### DISCUSSION TITLE
Quarkus extension written in Kotlin programming language

### DISCUSSION THREAD

**Discussion by b0ro** [quarkusio/quarkus#41312] [Q&A] [answered]
I'm trying to write Quarkus extension in Kotlin and I'm not sure how to start.

I've already rewritten one service from Java to Kotlin and I know it's a matter of adding kotlin-extension as described in https://quarkus.io/guides/kotlin.

But what should I do when I want to write Quarkus extension in that language? Should I add dependency to kotlin-extension as well (is it even possible)? Or should I just add kotlin maven plugin?

There's some documentation here https://kotlinlang.org/docs/maven.html#specify-compiler-options but I'm not sure which approach is correct for my case. Do you have any code examples I could base on?

I've checked Quarkus documentation if there's any info on how to write Quarkus extension in Kotlin but I found nothing.

[a comment by an automated account is not shown]

**Reply by b0ro** [reply]
Just let me know if I'm asking for something too obvious :)

**Reply by geoand** [reply]
This is a very good question.

The Quarkus team does not have any examples, but theoretically it should work if you are using Maven and just add the proper compiler configuration and Kotlin dependency

**Comment by mschorsch** [comment, marked as answer]
@b0ro https://github.com/quarkusio/quarkus/issues/40961#issuecomment-2156046994

**Reply by b0ro** [reply]
Nice, will check it out. Thx.

**Reply by b0ro** [reply]
Works like a charm :)
